### DISCUSSION TITLE
set_sodium_misuse and threading

### DISCUSSION THREAD

**Discussion by MichalPetryka** [jedisct1/libsodium#1205] [Q&A] [answered]
Is the set_sodium_misuse handler guaranteed to be called on the same thread that called a sodium function? I'd like to store info about what method fired it in TLS for logging purposes, which requires this guarantee.

**Comment by jedisct1** [comment, marked as answer]
Yes, it will already be called from the same thread.

But that handler should never be called, unless something really goes wrong.
